src/main/java/org/opensearch/knn/training/TrainingJob.java

```diff
@@ -182,6 +187,14 @@ public void run() {
                 KNNSettings.state().getSettingValue(KNNSettings.KNN_ALGO_PARAM_INDEX_THREAD_QTY)
             );
 
+            if (VectorDataType.BINARY.equals(model.getModelMetadata().getVectorDataType())) {
```

Comment:
@junqiu-lei Where are we validating if the data_type of training index is matching with that of the parameter passed in train model mapping ?

Reply:
I'm working on this part validate implementation, will update on the PR

Reply:
Added the validation part https://github.com/heemin32/k-NN/compare/90479ed9f4e0444fcf3bc43727fb7e66a6ffdd91..7dc64edffb98f81a1bec67d922536961c191cd51